src/devtools/client/debugger/src/utils/sourceVisualizations.ts

```diff
@@ -199,30 +230,41 @@ function getAlternateSourceIdForPosition(
       );
 }
 
-export function getAlternateSourceId(
+export function getAlternateSourceIdSuspense(
   client: ReplayClientInterface,
+  pauseId: PauseId,
   selectedSource: SourceDetails | null | undefined,
-  alternateSource: SourceDetails | null | undefined,
-  sourcesById: Dictionary<SourceDetails>,
-  position: CursorPosition
+  selectedFrameId: string | null,
+  position: CursorPosition,
+  sourcesState: SourcesState
 ): AlternateSourceResult {
-  if (alternateSource) {
-    return { sourceId: alternateSource.id };
-  }
   if (!selectedSource) {
     return { why: "no-source" };
   }
 
-  const uniqueAlternate = getUniqueAlternateSourceId(selectedSource, sourcesById);
+  let alternateSourceId = getAlternateSourceIdFromSelectedFrame(
+    pauseId,
+    selectedSource,
+    selectedFrameId,
+    sourcesState
+  );
+  if (alternateSourceId) {
+    return { sourceId: alternateSourceId };
+  }
+
+  const uniqueAlternate = getUniqueAlternateSourceId(
+    selectedSource,
+    sourcesState.sourceDetails.entities
+  );
   if (uniqueAlternate.sourceId || uniqueAlternate.why === "no-sourcemap") {
     return uniqueAlternate;
   }
```

Comment:
I'm not sure I understand how these lines make sense.

Reply:
If `getUniqueAlternateSourceId()` returns a `sourceId` or if it says that there is no sourcemap then we want to pass that on. If it says that there are multiple alternate sourceIds (`uniqueAlternate.why === "not-unique"`) then we want to carry on and call `getAlternateSourceIdForPositionSuspense()`